### ISSUE TITLE
Solving classical puzzles by a quantum computer

### ISSUE THREAD

**Issue by a-matsuo** [qiskit-community/qiskit-hackathon-korea-21#4]
# Abstract
We can solve classical puzzles by a quantum computer. For example, they can be "8 queens", "Sudoku" and so on. In this project, we can decide what puzzle we want to solve. So please think and discuss what puzzle we will solve with other team members. Also, we can choose a solving method from Grover's algorithm, VQE or QAOA. This is an [example of solving the "Asteroid" puzzle with Grover's algorithm](https://github.com/qiskit-community/IBMQuantumChallenge2020/blob/main/solutions/week-3/ex_3_solution_en.ipynb) from the last year's IBM Quantum Challenge. For VQE and QAOA, first, we need to formulate a puzzle that we want to solve with an optimization model like [this example](https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/optimization/1_quadratic_program.ipynb). It looks tricky, but actually it's not. You just need a basic knowledge of math.

ToDo:
- [ ] Discuss and decide what puzzle we want to solve. (We don't need to chose just one. We can try multiple puzzles)
- [ ] Discuss and decide what method we want to use. **Note that each method requires different thing we need to make.** For Grover's algorithm, we need to make an oracle for the puzzle. For VQE/QAOA, we need to make an optimization model for the puzzle.
  - Grover's algorithm (need an oracle)
  - VQE/QAOA (need an optimization model)

# Members

 - @githubhandle 
 - @githubhandle - Slack: `@slackhandle` email: `example@example.com`
 - Qiskit Coach: @a-matsuo  

# Deliverable

# GitHub repo

**Comment by hongildong2** [comment]
Sounds interesting! how can i join this project?

**Comment by icepolarizer** [comment]
I'm a mentor, but I'm highly interested in this kind of topic. Perhaps I can even learn while helping. Please tell me if there's anything I can help.

**Comment by starktech23** [comment]
@hongildong2 you can start by finding some team members to work together on this :)

**Comment by Yunjeong-Quim** [comment]
I am a beginner and new to quantum computing, so I am quite afraid that I might be helpless.. haha
Though I love Puzzles like sudoku, and I am very interested in quantum Algorithm. 
May I join this project?

**Comment by zero-or-one** [comment]
Hi, I am a big fan of puzzle games and have a good knowledge in programming and math. However I am don't know Korean language well so it may be hard for me to communicate.
Can I join this project?

**Comment by a-matsuo** [comment]
This is just an idea. If you can find some traditional Korean puzzle or your own country puzzle, it can be interesting :D

**Comment by Yunjeong-Quim** [comment]
> This is just an idea. If you can find some traditional Korean puzzle or your own country puzzle, it can be interesting :D

great! I will try to think of some of traditional puzzle like Yutnori or Baduk.
Let me think about how Grover's algorithm can be applied to this... And let you know before 2/18 12am(KST)

**Comment by starktech23** [comment]
@uno-b1-c3 Yes you can join! The primary medium of communication and presentation will be in English so language won't be a barrier to your participation.

**Comment by Han-JaeHoon** [comment]
Is there any group I can join?

**Comment by veenaiyuri** [comment]
@Han-JaeHoon Sure! If you are interested in this project, I can assign you. Please remember that your team number is #4. So you can join the team room tomorrow morning.

**Comment by starktech23** [comment]
@uno-b1-c3 @Han-JaeHoon @Yunjeong-Quim @hongildong2 are you 4 joining as a team to work on this project?

**Comment by Han-JaeHoon** [comment]
Oh, I'm sorry, but I'm already on another team. I'll have to leave out of this issue because it's probably not possible to join the team in duplicate.

**Comment by zero-or-one** [comment]
I am in

**Comment by x2ever** [comment]
Are you still need a additional member? I want to join here!

**Comment by a-matsuo** [comment]
Sure, you can join. Come to break room #4 in the zoom
